task/mongo-optimization-task.js

```diff
@@ -243,22 +280,33 @@ async function task_3_1(db) {
         {
             "$lookup" : {
                 "from" : "clientCriteria",
-                "localField" : "criteria_value",
-                "foreignField" : "value",
+                "let" : {"value1": "$criteria_value"},
+                "pipeline": [{
+                    "$match": {
+                        "versions.initiativeId": ObjectId("58af4da0b310d92314627290")
```

Comment:
you can make it in one `$match`. Also try to use compound index here and `$limit` at the end of the pipeline as logic of the query expect only one item in the final array.